### PR TITLE
require cycle warning removed by extracting dbcontext to its own file

### PR DESCRIPTION
**Pull request by sj19330**
moved the db context into its own file which is then separately imported into the app.js file and the track weight/food pages, removing the requirement cycle that we discussed!!

everything is looking good on my end, should be good to merge with no problems, not much has been changed

### CODE REVIEW

File: App.js
Comment (jxz12) [inline App.js:1]: You should also factor out the `DBContext.Provider` tag into a separate component, and then you can put the `initialiseDatabase` function into that file too.

Here is an article with an example: https://javascript.works-hub.com/learn/building-with-react-context-provider-pattern-1af4b

**Review by jxz12 — CHANGES_REQUESTED** [review]
please factor out the provider as well as the context into a separate file, then the database initialisation will be nicely abstracted away into its own component

**Review by sj19330 — COMMENTED** [review]
Refactored so the provideor, loading and database initialization is all in the DbContext.js file instead of the App.js. Tested database with the log weight and log food pages and all is working well : )

File: DbContext.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import { useState, useEffect, createContext } from "react";
+import { Text, View } from "react-native";
+import * as SQLite from "expo-sqlite";
+
+export const DBContext = createContext(undefined);
+
+export default function DataBaseContext({ children }) {
```
Comment (jxz12) [inline DbContext.js]: This should be named `Provider` instead of Context, so `DBProvider` would be better.

**Review by jxz12 — APPROVED** [review]
good work, feel free to merge after renaming the provider class!

**Pull request merged**